apps/chat/src/components/AppsEditor/GeneralInfoView/form.ts

```diff
@@ -0,0 +1,204 @@
+import { ApiUtils } from '@/src/utils/server/api';
+
+import { ApiDetailedApplicationTypeSchema } from '@/src/types/application-type-schema';
+import {
+  ApiApplicationResponseDefault,
+  ApplicationStatus,
+  ApplicationType,
+  CustomApplicationModel,
+} from '@/src/types/applications';
+import { EntityType } from '@/src/types/common';
+import { QuickAppConfig } from '@/src/types/quick-apps';
+
+import {
+  FEATURES_ENDPOINTS,
+  FEATURES_ENDPOINTS_DEFAULT_VALUES,
+  FEATURES_ENDPOINTS_NAMES,
+} from '@/src/constants/applications';
+import { DEFAULT_APPLICATION_NAME } from '@/src/constants/default-ui-settings';
+import { DEFAULT_VERSION } from '@/src/constants/public';
+
+import { DynamicField } from '../../Common/Forms/DynamicFormFields';
+
+import { ShareEntity } from '@epam/ai-dial-shared';
+
+export interface ApplicationGeneralInfoFormData {
+  name: string;
+  version: string;
+  iconUrl: string;
+  description: string;
+  topics: string[];
+  completionUrl: string;
+  id: string;
+  reference: string;
+  applicationProperties?: Record<string, unknown> | QuickAppConfig | null;
+  inputAttachmentTypes?: string[];
+  maxInputAttachments?: number;
+  features?: string | null;
+  env?: DynamicField[];
+  sources?: string;
+  sourceFiles?: string[];
+  runtime?: string;
+  endpoints?: DynamicField[];
+  functionStatus?: ApplicationStatus;
+}
+
+type Options<T extends Path<ApplicationGeneralInfoFormData>> = Omit<
+  RegisterOptions<ApplicationGeneralInfoFormData, T>,
+  'disabled' | 'valueAsNumber' | 'valueAsDate'
+>;
+
+type Validators = {
+  [K in keyof ApplicationGeneralInfoFormData]?: Options<K>;
+};
+
+export const getDefaultValues = (
+  applicationData?: ApiApplicationResponseDefault,
+  bucket?: string,
+  pythonVersion?: string,
+  models?: ShareEntity[],
+): ApplicationGeneralInfoFormData => {
+  return {
+    name:
+      applicationData?.display_name ??
+      getNextDefaultName(DEFAULT_APPLICATION_NAME, models ?? [], 0, true),
+    version: applicationData?.display_version ?? DEFAULT_VERSION,
+    iconUrl: applicationData?.icon_url ?? '',
+    description: applicationData?.description ?? '',
+    topics: applicationData?.description_keywords ?? [],
+    id: applicationData?.name ? decodeURIComponent(applicationData?.name) : '',
+    reference: applicationData?.reference ?? '',
+    //schema type application properties
+    applicationProperties: applicationData?.application_properties,
+    //custom application properties
+    completionUrl: applicationData?.endpoint ?? '',
+    inputAttachmentTypes: applicationData?.input_attachment_types,
+    maxInputAttachments: applicationData?.max_input_attachments,
+    features: safeStringifyApplicationFeatures(applicationData?.features),
+    //code app application properties
+    sources: applicationData?.function?.source_folder
+      ? ApiUtils.decodeApiUrl(applicationData.function.source_folder)
+      : `files/${bucket}/appdata`,
```

Comment:
Source folder should be empty as default. We have 'examples' if we don't have files in the selected directory. If we click on some example application now, we will create these files in the /appdata directory which is not supposed to be used that way.

Reply:
I updated